src/components/ProductList/Filter.js

```diff
@@ -0,0 +1,196 @@
+import React, { useState } from "react";
+import "./Filter.css";
+import { useFilter } from "../../context/filter-context";
+export default function Filter() {
+  const [showFilter, setShowFilter] = useState(false);
+  const {
+    state: { priceSort, category, ratingSort, range },
+    dispatch,
+  } = useFilter();
+
+  return (
+    <>
+      <button onClick={() => setShowFilter(() => !showFilter)} class="filter">
+        <i class="fas fa-sort"></i> Filters
+      </button>
+      {showFilter ? (
+        <div className="filter-div">
+          <div>
+            <h3 className="filter-heading">
+              <u>Price</u>
+              <button
+                onClick={() => dispatch({ type: "CLEAR_FILTER" })}
+                className="clear-filter"
+              >
+                Clear
+              </button>
+            </h3>
+            <input
+              type="radio"
+              checked={priceSort === "low-to-high-price"}
+              onChange={() =>
+                dispatch({
+                  type: "PRICE_SORT",
+                  payload: "low-to-high-price",
+                })
+              }
+              value="low-high-price"
+              name="price-sort"
+              id="low-high-price"
+            />
+            <label htmlFor="price-sort" className="checkbox">
+              Price: low to high
+            </label>
```

Comment:
why not wrap the `input` element inside the `label`? , is there any specific reason